src/sentry_attachment.c

```diff
@@ -43,33 +86,45 @@ sentry__attachments_free(sentry_attachment_t *attachments)
         sentry_attachment_t *attachment = it;
         it = attachment->next;
 
-        attachment_free(attachment);
+        sentry__attachment_free(attachment);
     }
 }
 
+/**
+ * Compares attachments for equality to avoid adding duplicates and for removal.
+ *
+ * File attachments are considered equal if the paths are equal. Byte buffer
+ * attachments are only considered equal to themselves - they are not required
+ * to have unique filenames, and the bytes are not compared.
```

Comment:
Upon re-reading this comment, the behavior still warrants a note in the public header documentation.

It should be noted that

* Neither the `filename` nor the `bytes` parameters' content is part of the attachment identity for bytes-based attachments (whereas with file-based attachments, the `filename` is the identity).
* We nonetheless recommend using unique filenames (even if the public API does not enforce them), because they will be easier to differentiate in the Web UI and (seqway to) `crashpad` also requires unique filenames for the multipart upload (continue with crashpad note).

Reply:
Now that I'm re-reading it again myself, I'm not sure why I decided to use `attach_eq` even for removal. The pointer comparison should be enough for removal and not cause any surprises.